src/js/components/modals/ServiceFormModal.js

```diff
@@ -258,13 +270,15 @@ class ServiceFormModal extends mixin(StoreMixin) {
     }
 
     if (this.state.jsonMode) {
-      let jsonDefinition = this.state.jsonDefinition;
-      marathonAction(JSON.parse(jsonDefinition), this.state.force);
+      let jsonDefinition = JSON.parse(this.state.jsonDefinition);
+      jsonDefinition = cleanJSONdefinition(jsonDefinition);
+      marathonAction(jsonDefinition, this.state.force);
+      jsonDefinition = JSON.stringify(jsonDefinition, null, 2);
```

Comment:
We're probably able to drastically simplify the whole `handleSubmit` process. I wonder if we should do this in a separate PR. I'd love if we could simplify it in the same way we've did it for the jobs (https://github.com/dcos/dcos-ui/blob/master/src/js/components/modals/JobFormModal.js#L157).